File: app/src/main/java/ru/profapp/RanobeReader/DAO/DateConverter.java
package ru.profapp.RanobeReader.DAO;

import android.arch.persistence.room.TypeConverter;

import java.util.Date;

/**
 * Created by Ruslan on 09.02.2018.
 */

public class DateConverter {

    @TypeConverter
    public static Date toDate(long dateLong) {
        if (dateLong == 0) {
            return null;
        }
        return new Date(dateLong);
    }

    @TypeConverter
    public static long fromDate(Date date) {
        if (date == null) {
            return 0;
        }
        return date.getTime();
    }
}

File: app/src/main/java/ru/profapp/RanobeReaderTest/DAO/DateConverter.kt
package ru.profapp.RanobeReaderTest.DAO

import androidx.room.TypeConverter
import java.util.*

/**
 * Created by Ruslan on 09.02.2018.
 */

class DateConverter {

    @TypeConverter
    fun toDate(value: Long?): Date? {
        return if (value == null) null else Date(value)
    }

    @TypeConverter
    fun fromDate(date: Date?): Long? {
        return date?.time
    }
}
